xbout/load.py

```diff
@@ -555,13 +556,24 @@ def _open_grid(datapath, chunks, keep_xboundaries, keep_yboundaries, mxg=2):
     boundaries to deal with different conventions in a BOUT grid file.
     """
 
+    acceptable_dims = ['x', 'y', 'z']
+
+    # Passing 'chunks' with dimensions that are not present in the dataset causes an
+    # error. A gridfile will be missing 't' and may be missing 'z' dimensions that dump
+    # files have, so we must remove them from 'chunks'.
+    unrecognised_chunk_dims = list(set(chunks.keys()) - set(acceptable_dims))
+    chunks = copy(chunks)
+    for dim in unrecognised_chunk_dims:
+        del chunks[dim]
+
     gridfilepath = Path(datapath)
-    grid = xr.open_dataset(gridfilepath, engine=_check_filetype(gridfilepath),
-                           chunks=chunks)
+    grid = xr.open_dataset(gridfilepath, engine=_check_filetype(gridfilepath))
+    if 'z' in chunks and 'z' not in grid.dims:
+        del chunks['z']
+    grid.chunk(chunks)
```

Comment:
`ds.chunk(chunks)` appears to have a [return value](http://xarray.pydata.org/en/stable/generated/xarray.Dataset.chunk.html#xarray.Dataset.chunk), so this should be `grid = grid.chunks(chunks)`